test/msw-api/life-cycle-events/life-cycle-events.mocks.ts

```diff
@@ -0,0 +1,39 @@
+import { rest, setupWorker } from 'msw'
+
+const worker = setupWorker(
+  rest.get('/user', (req, res, ctx) => {
+    return res(ctx.text('response-body'))
+  }),
+  rest.post('/no-response', (req, res, ctx) => {
+    return
+  }),
+)
+
+worker.on('request:start', (req) => {
+  console.warn(`[request:start] ${req.method} ${req.url.href} ${req.id}`)
+})
+
+worker.on('request:match', (req) => {
+  console.warn(`[request:match] ${req.method} ${req.url.href} ${req.id}`)
+})
+
+worker.on('request:unhandled', (req) => {
+  console.warn(`[request:unhandled] ${req.method} ${req.url.href} ${req.id}`)
+})
+
+worker.on('request:end', (req) => {
+  console.warn(`[request:end] ${req.method} ${req.url.href} ${req.id}`)
+})
+
+worker.on('response:mocked', async (res, reqId) => {
+  const body = await res.text()
```

Comment:
Will a user always use `await res.text()`? I know this is produced by:

`const response = new Response(payload.body, payload)`

But what's the full use-case look like here? I can't really tell from the test what _should_ be going on. Ideally, I'd imagine we would recreate the original response with it's content-type, then provide an unwrap utility so someone can work with the object or text without trying to reconstruct it?

I'll poke around a bit in the run time... maybe I'm missing something :)

Reply:
It's up to the consumer to handle the response in this listener. The intention is to return a cloned original/mocked responses here so that the consumer may operate on them in any way needed. That includes constructing a response body (`res.text`/`res.json`/etc.), accessing headers, etc. There is nothing that _should_ happen precisely, as it's up to the consumer to utilize these listeners. All these listeners are read-only and don't affect what happens or happened in the library. 

I find the native `Response` instance to be more suitable here than a custom abstraction, because:

1. It's more familiar (`fetch()` returns the same instance).
1. It gives more control and is quite convenient to handle.
1. It makes `response:mocked` and `response:bypass` behave homogenously, although the source of response differs for those two events.